arabic/unpacked/tex.js

```diff
@@ -0,0 +1,205 @@
+          token.data[0].data[0] = mapped;
+          token = token.With({
+            fontLang: 'ar'
+          });
+        }
+
+        return this.flipHorizontal(token);
+      },
+      arabicOperator: function (token) {
+        var operatorsMap = MathJax.Hub.config.Arabic.operatorsMap;
+        var text = token.data[0].data[0];
+        var mapped = text;
+
+        Object.keys(operatorsMap).forEach(function (enOperator) {
+          var regex = new RegExp('' + enOperator, 'g');
+          var arOperator = operatorsMap[enOperator];
+          mapped = mapped.replace(regex, arOperator);
+        });
+
+        if (mapped !== text) {
+          token = this.flipHorizontal(token).With({
+            fontLang: 'ar'
+          });
+
+          token.data[0].data[0] = mapped;
+        }
+
+        return token;
+      },
+      _getArgumentMML: function (name) {
+        //  returns an argument that is a single MathML element
+        //  (in an mrow if necessary)
+        //
+        //  This functions has been copied here from extensions/TeX/HTML.js, to avoid
+        //  adding a dependency.
+        //
+        //  TODO: Consider importing (as a dependency) this from HTML.js instead!
+        var arg = this.ParseArg(name);
+        if (arg.inferred && arg.data.length == 1)
+          {arg = arg.data[0]} else {delete arg.inferred}
+        return arg;
+      },
+      mmlToken: function (token) {
+        // TODO: Check for possible incompatibility with boldsymbol extension
+        var parsedToken = texParseMMLToken.apply(this, [token]);
+
+        if ('ar' === this.stack.env.lang) {
+          this.markArabicToken(parsedToken);
+        }
+
+        return parsedToken;
+      },
+      markArabicToken: function (token) {
+        if ('mn' === token.type) {
+          return this.arabicNumber(token);
+        } else if ('mi' === token.type) {
+          return this.arabicIdentifier(token);
+        } else if ('mo' === token.type) {
+          return this.arabicOperator(token);
+        }
+
+        return token;
+      },
+      AlignedArray: function () {
+        // Helper to Arabize the matrices, arrays and piecewise functions.
+        var array = texParseAlignedArray.apply(this, arguments);
+        var _this = this;
+
+        if ('ar' === this.stack.env.lang) {
+          var arrayEndTable = array.EndTable;
+          array.EndTable = function () {
+            var retVal = arrayEndTable.apply(this, arguments);
+
+            // First level, iterate over the rows
+            array.table.forEach(function (row, rowIndex) {
+              // Second level, iterate over the columns
+              row.data.forEach(function (cell, colIndex) {
+                // Third level, iterate over the cell content
+                cell.data[0].data.forEach(function (token) {
+                  // Skip the first element of the matrix to solve a bug,
+                  // not sure why it is there.
+                  if (rowIndex || colIndex) {
```

Comment:
Can you describe the bug that you are trying to avoid?  I'm not sure what it might be, unless it is the possible empty cells.  This can happen when there are labels, for example (e.g., with `\tag{}` or auto numbered equations).  Have you tested with those?